src/target/linux.rs

```diff
@@ -221,6 +306,192 @@ impl LinuxTarget {
 
         Ok(tasks)
     }
+
+    #[cfg(target_arch = "x86_64")]
+    pub fn set_hardware_breakpoint(
+        &mut self,
+        breakpoint: HardwareBreakpoint,
+    ) -> Result<usize, Box<dyn std::error::Error>> {
+        #[cfg(target_arch = "x86_64")]
+        {
+            let index = if let Some(empty) = self.find_empty_watchpoint() {
+                empty
+            } else {
+                return Err(Box::new(HardwareBreakpointError::NoEmptyWatchpoint));
+            };
+
+            let rw_bits: u64 = match breakpoint.typ {
+                HardwareBreakpointType::Execute => 0b00,
+                HardwareBreakpointType::Read => 0b11,
+                HardwareBreakpointType::ReadWrite => 0b11,
+                HardwareBreakpointType::Write => 0b01,
+            } << 16 + index * 4;
+
+            let size_bits: u64 = (breakpoint.size as u64) << (18 + index * 4);
+
+            let enable_bit: u64 = 1 << (2 * index);
+
+            let bit_mask: u64 = (0b11 << (2 * index)) | (0b1111 << (16 + 4 * index));
```

Comment:
This would also work good as a function in `HardwareBreakpoint`, e.g.:
```rust
impl HardwareBreakpoint {
    const fn bit_mask(index: usize) -> u64 {
        (0b11 << (2 * index)) | (0b1111 << (16 + 4 * index))
    }
}
```
then this line can be replaced with
```suggestion
            let bit_mask = HardwareBreakpoint::bit_mask(index);
```